src/Deinterleave.cpp

```diff
@@ -658,6 +658,12 @@ class Interleaver : public IRMutator {
             return Stmt();
         }
 
+        // Too many stores and lanes to represent in a single vector
+        // type.
+        if (stores.size() * lanes > std::numeric_limits<decltype(halide_type_t{}.lanes)>::max()) {
```

Comment:
This only works when ```stores.size()``` has a larger type than ```lanes```. (Or int is bigger than the type of each.) Which is a reasonable assumption, but is a bit subtle and gets to the general problem of how to test for overflow. Feel free to ignore the comment, but I might consider either comparing against a divide or using an explicit cast to 64-bit.